src/rand.rs

```diff
@@ -242,6 +241,42 @@ mod sysrand_chunk {
     }
 }
 
+#[cfg(target_os = "freebsd")]
+mod sysrand_chunk {
+    use crate::{c, error};
+
+    #[inline]
+    pub fn chunk(dest: &mut [u8]) -> Result<usize, error::Unspecified> {
+        use libc::c_int;
+
+        // See kern/syscalls.master, or the generated `SYS_getrandom` in #include <sys/syscall.h>
+
+        const SYS_GETRANDOM: c_int = 563;
+
+        // As long as we're using libc, it would be preferable if the libc crate provided
+        // getrandom() directly.  It would also be preferable to use __syscall rather than
+        // syscall().  However, at this time the Rust libc crate provides neither of these
+        // functions on FreeBSD.
+
+        // Clamp request size to INT_MAX due to limitation of the return type of libc::syscall()
+        // (c_int).
+        let chunk_len: c::size_t = dest.len().min(c_int::max_value() as usize);
+        let r = unsafe { libc::syscall(SYS_GETRANDOM, dest.as_mut_ptr(), chunk_len, 0) };
```

Comment:
Does FreeBSD have a stable syscall ABI, or is it one of those platforms where only libc has a stable ABI?

Reply:
I could only find https://wiki.freebsd.org/Releng/ABI, which says:

> We also try to maintain ABI compatibility across .0 releases, but they are not strictly enforced except for libraries that already implements versioned symbols.

Which I read as best effort, but not guaranteed.

Reply:
I'll see if I can add it to libc.

Reply:
PR for libc https://github.com/rust-lang/libc/pull/1982.

Reply:
It's one of those platforms where only libc has a stable ABI.  The plan is to provide a stable `libsystem` ABI at some point in the near future for non-C programming languages, but I'm afraid we're not there yet.

Reply:
> PR for libc rust-lang/libc#1982.

Great. It seems like we could just call `getrandom` directly from *ring* through FFI, without going through the libc crate, couldn't we? It seems like maybe we should go that route.

Reply:
Sure, you could link FreeBSD libc directly in *ring* without using the libc crate.